base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java

```diff
@@ -82,6 +70,63 @@ public void init() throws Exception {
         }
     }
 
+    /**
+     * This method will create the initial connection, validate
+     * the current connection, or reestablish the connection if
+     * it's closed.
+     *
+     * This method should only be called by methods implementing
+     * ACMEDatabase.
+     *
+     * TODO: Use connection pool.
+     */
+    public void connect() throws Exception {
+
+        if (connection == null) { // create the initial connection
+
+            logger.info("Connecting to " + url);
+            connection = DriverManager.getConnection(url, info);
+
+            DatabaseMetaData md = connection.getMetaData();
+            ResultSet rs = null;
+
+            try {
```

Comment:
Is there a specific reason we're attempting to get the list of tables? Prove that the connection works? Make sure it is set up (should we be validating the results?).

Reply:
In the original code the connection was created on startup, and this code allowed the admin to verify visually that the database was ready as well. In the new code the connection is created on demand, and now we have auto-recovery too, so this code is not needed anymore. I'll remove it before merging. Thanks!